src/main/java/io/fabric8/maven/XMLFormat.java

```diff
@@ -0,0 +1,105 @@
+package io.fabric8.maven;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringWriter;
+import java.io.UncheckedIOException;
+
+import org.jdom2.Document;
+import org.jdom2.JDOMException;
+import org.jdom2.input.SAXBuilder;
+import org.jdom2.output.Format;
+import org.jdom2.output.XMLOutputter;
+
+/**
+ * Output format for an XML document
+ * <p>
+ * This experimental class is a simple wrapper around JDOM2's {@link XMLOutputter} to provide a simple way to format XML
+ */
+public class XMLFormat {
+
+    private final int indent;
+
+    private final boolean insertLineBreakBetweenMajorSections;
+
+    private XMLFormat(Builder builder) {
+        this.indent = builder.indent;
+        this.insertLineBreakBetweenMajorSections = builder.insertLineBreakBetweenMajorSections;
+    }
+
+    /**
+     * @return the number of spaces to use for indentation
+     */
+    public int getIndent() {
+        return indent;
+    }
+
+    /**
+     * @return true if a line break should be inserted between major sections
+     */
+    public boolean isInsertLineBreakBetweenMajorSections() {
+        return insertLineBreakBetweenMajorSections;
+    }
+
+    /**
+     * Format the XML from the given reader
+     *
+     * @param reader the reader
+     * @return the formatted XML
+     */
+    public String format(Reader reader) {
+        Document document;
+        try {
+            document = new SAXBuilder().build(reader);
+            XMLOutputter xmlOutputter = new XMLOutputter();
+            String indentStr = " ".repeat(indent);
+            Format format = Format.getPrettyFormat()
+                    .setIndent(indentStr);
+            xmlOutputter.setFormat(format);
+            StringWriter sw = new StringWriter();
+            xmlOutputter.output(document, sw);
+            String result = sw.toString();
+            if (insertLineBreakBetweenMajorSections) {
+                // Using a regex to insert a line break between major sections
+                return result.replaceAll("^" + indentStr + "<[^/]", ">\n<");
```

Comment:
What does `">\n<"` do exactly? I don't know this syntax?

Reply:
Ignore that, I was focusing on the API first